src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -0,0 +1,503 @@
+
+const getCategorizedVersions = (
+	deviceTypeOsVersions: OsVersionsByDeviceType,
+	deviceType: DownloadImageFormModel['deviceType'],
+	osType: string | null,
+) => {
+	const osVersions = deviceTypeOsVersions[deviceType.slug] ?? [];
+	const deviceOsVersions = osType
+		? osVersions.filter((osVersion) => osVersion.osType === osType)
+		: osVersions;
+
+	const selectionOpts = transformVersions(deviceOsVersions);
+	const preferredSelectionOpts = getPreferredVersionOpts(selectionOpts);
+
+	return {
+		selectionOpts,
+		preferredSelectionOpts,
+	};
+};
+
+const lineMap = {
+	next: 'info.light',
+	current: 'success.light',
+	sunset: 'warning.light',
+	outdated: 'error.light',
+};
+
+interface ImageFormProps {
+	applicationId: number;
+	releaseId?: number;
+	compatibleDeviceTypes: DeviceType[] | undefined;
+	osVersions: OsVersionsByDeviceType;
+	osType: string | null;
+	osTypes: string[];
+	formElement: any;
+	downloadUrl: string;
+	authToken: string | undefined;
+	isInitialDefault?: boolean;
+	model: DownloadImageFormModel;
+	hasEsrVersions?: boolean;
+	onSelectedOsTypeChange: (osType: string) => void;
+	onSelectedDeviceTypeChange: (deviceType: DeviceType) => void;
+	onChange: (
+		key: keyof DownloadImageFormModel,
+		value: DownloadImageFormModel[keyof DownloadImageFormModel],
+	) => void;
+}
+
+export const ImageForm: React.FC<ImageFormProps> = memo(
+	({
+		compatibleDeviceTypes,
+		osVersions,
+		isInitialDefault,
+		osType,
+		osTypes,
+		hasEsrVersions,
+		model,
+		formElement,
+		downloadUrl,
+		applicationId,
+		authToken,
+		onSelectedOsTypeChange,
+		onSelectedDeviceTypeChange,
+		onChange,
+	}) => {
+		const [showAdvancedSettings, setShowAdvancedSettings] = useState(false);
+		const [version, setVersion] = useState<
+			VersionSelectionOptions | undefined
+		>();
+		const [variant, setVariant] = useState<BuildVariant>('prod');
+		const [showAllVersions, setShowAllVersions] = useState(false);
+		const { selectionOpts, preferredSelectionOpts } = useMemo(
+			() => getCategorizedVersions(osVersions, model.deviceType, osType),
+			[osVersions, model.deviceType, osType],
+		);
+		const versionSelectionOpts = useMemo(
+			() => (showAllVersions ? selectionOpts : preferredSelectionOpts),
+			[preferredSelectionOpts, selectionOpts, showAllVersions],
+		);
+		const showAllVersionsToggle = useMemo(
+			() => preferredSelectionOpts.length < selectionOpts.length,
+			[preferredSelectionOpts.length, selectionOpts.length],
+		);
+
+		const handleShowAllVersions = (e: React.ChangeEvent<HTMLInputElement>) => {
+			const isChecked = e.target.checked;
+			setShowAllVersions(isChecked);
+
+			if (isChecked || !version) {
+				return;
+			}
+
+			const selectedValueIsPreferred = preferredSelectionOpts.some(
+				(ver) => ver.value === version.value,
+			);
+			if (selectedValueIsPreferred) {
+				return;
+			}
+
+			const preferred =
+				preferredSelectionOpts.find((ver) => ver.isRecommended) ??
+				preferredSelectionOpts?.[0];
```

Comment:
Optional chaining in one but not the other? Do we need the optional chain or not? If yes I would expect it on both, if no I would expect it on neither